crates/node/src/commands/run.rs

```diff
@@ -2,15 +2,48 @@ use std::path::PathBuf;
 
 use madara_runtime::SealingMode;
 use mc_data_availability::DaLayer;
-use mc_l1_messages::config::L1MessagesWorkerConfig;
-use mc_l1_messages::error::L1MessagesConfigError;
+use mc_l1_messages::config::{L1MessagesWorkerConfig, L1MessagesWorkerConfigError};
 use sc_cli::{Result, RpcMethods, RunCmd, SubstrateCli};
 use sc_service::BasePath;
 use serde::{Deserialize, Serialize};
 
 use crate::cli::Cli;
 use crate::service;
 
+#[derive(Debug, Clone, clap::Args)]
+pub struct L1MessagesParams {
+    /// Ethereum Provider (Node) Url
+    #[clap(
+        long,
+        requires="l1_contract_address",
+        value_hint=clap::ValueHint::Url,
+    )]
+    pub provider_url: String,
+
+    /// L1 Contract Address
+    #[clap(
+        long,
+        requires="provider_url",
+        value_hint=clap::ValueHint::Other,
```

Comment:
No need for those two requires. If they are not Option<T>, I think that they are both required